### PR TITLE
[PAGOPA-1946] feat: routing whitelisted payment types on WISP dismantling

### PR DESCRIPTION
**Pull request by andrea-deri**
#### Apply
In the following order the target applies were executed with the following command:  
 - For `nodo-app` domain  
```
sh terraform.sh apply weu-ENV \
    --target=azurerm_api_management_named_value.wisp_nodoinviarpt_paymenttype_whitelist_named_value
```
 - For `core` domain  
```
sh terraform.sh apply ENV \
    --target=azapi_resource.decoupler_algorithm \
    --target=terraform_data.sha256_decoupler_algorithm
```

### List of changes
 - created new named value for include WISP dismantling's accepted payment types
 - updated policy in order to include the checks to be executed for RPT's payment types on WISP dismantling

### Motivation and context
This change is required in order to add another filter for WISP dismantling

### Type of changes
- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?
- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result
- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/core/api_product/nodo_pagamenti_api/decoupler/decoupler-algorithm.xml [hunk 1]
```diff
@@ -331,9 +333,21 @@ new JProperty("ttl", 1)
...
+                            // if is whitelisted and it is a 'nodoInviaRPT'
+                            bool isNodoInviaRPT = "nodoInviaRPT".Equals((string)context.Variables["soapAction"]);
+                            if (isWhitelisted && isNodoInviaRPT) {
+                                byte[] rptAsBytearray = Convert.FromBase64String((string) primitive.Descendants("rpt").FirstOrDefault());
+                                string rptAsString = System.Text.Encoding.UTF8.GetString(rptAsBytearray);
+                                XElement rpt = XElement.Parse(rptAsString);
+                                string rptPaymentType = rpt.Descendants(rpt.Name.Namespace + "tipoVersamento").FirstOrDefault()?.Value;
+                                string[] paymentTypes =  ((string) context.Variables["wisp_whitelisted_nodoinviarpt_paymenttypes"]).Split(',');
+                                isWhitelisted = paymentTypes.Contains(rptPaymentType);
+                            }
```
Comment (pasqualespica) [inline src/core/api_product/nodo_pagamenti_api/decoupler/decoupler-algorithm.xml:346]: - shift up this code before `isWhitelisted = (brokersWhitelisted ...`
- set vars type `isValidNodoInviaRptOrNodoInviaCarrelloRPT` with something like that 
`isValidNodoInviaRptOrNodoInviaCarrelloRPT` = `isNodoInviaCarrelloRPT` or `isNodoInviaRptWithAllowedPaymentType`

**Review by pasqualespica — CHANGES_REQUESTED** [review]
@andrea-deri snipped code it's right.
but for readability I suggest a small refactoring as suggested in the comment.
Do you agree ?

File: src/core/api_product/nodo_pagamenti_api/decoupler/decoupler-algorithm.xml (hunk 1 above)
Comment (andrea-deri) [reply src/core/api_product/nodo_pagamenti_api/decoupler/decoupler-algorithm.xml:346]: Yes, it's a good idea. I will apply your suggestions on the code. 🔝

Comment (andrea-deri) [reply src/core/api_product/nodo_pagamenti_api/decoupler/decoupler-algorithm.xml:343]: I have updated the code following your suggestion. Is it correct and is it what you meant in your comment?

**Review by pasqualespica — APPROVED** [review]
👍

**Pull request merged**